test/Microsoft.ML.Tests/FeatureContributionTests.cs

```diff
@@ -30,15 +32,47 @@ public FeatureContributionTests(ITestOutputHelper output) : base(output)
         {
         }
 
+        [Fact]
+        public void TestOrdinaryLeastSquares()
+        {
+            var expectedValues = new List<float[]> {
+                new float[4] { 0.06319684F, 1, 0.1386623F, 4.46209469E-06F },
+                new float[4] { 0.03841561F, 1, 0.1633037F, 2.68303256E-06F },
+                new float[4] { 0.12006103F, 1, 0.254072F, 1.18671605E-05F },
+                new float[4] { 0.20861618F, 0.99999994F, 0.407312155F, 6.963478E-05F },
+                new float[4] { 0.024050576F, 0.99999994F, 0.31106182F, 8.456762E-06F }, };
+
+            TestFeatureImportance(ML.Regression.Trainers.OrdinaryLeastSquares(), expectedValues);
+        }
+
+        [Fact]
+        public void TestGam()
+        {
+            // Index 1: Most important feature
+            // Index 3: Random feature
+            var expectedValues = new List<float[]> {
+                new float[4] { 0.08439296F, 1F, 0.1442171F, -0.001832674F },
+                new float[4] { -0.07902145F, -1F, -0.01937493F, 0.02314214F },
+                new float[4] { 0.04072217F, -1F, 0.01370963F, -0.00197823F },
+                new float[4] { -0.02197981F, -1F, -0.1051985F, -0.004131221F },
+                new float[4] { -0.1072952F, -1F, 0.1284171F, -0.002337188F }, };
+
+            TestFeatureImportance(ML.Regression.Trainers.GeneralizedAdditiveModels(), expectedValues, 5);
+        }
+
         /// <summary>
         /// Features: x1, x2, x3, xRand; y = 10*x1 + 20x2 + 5.5x3 + e, xRand- random, Label y is dependant on xRand.
         /// Test verifies that feature contribution scores are outputted along with a score for predicted data. 
         /// </summary>
-        [Fact]
-        public void TestFeatureImportance()
+        private void TestFeatureImportance(
```

Comment:
`TestFeatureContribution`

Reply:
renamed

---
In reply to: [240855403](https://github.com/dotnet/machinelearning/pull/1856#discussion_r240855403) [](ancestors = 240855403)